src/entity/spatial/Polygon.cpp

```diff
@@ -60,12 +60,9 @@ void Polygon::setCoordinates(const geom::CoordinateSequence& seq)
 
     std::vector< geom::Geometry* > holes;
 
-    auto polygon = factory_->createPolygon(*ring, holes);   //copy ring to avoid ownership issues
+    auto polygon = factory_->createPolygon(ring, new std::vector< geom::Geometry* >());
```

Comment:
This renders the `holes`-vector  redundant.
So either remove the local vector, or make it a pointer and use it in the method call for readability. :)